### DISCUSSION TITLE
WebSocket API

### DISCUSSION THREAD

**Discussion by zero734kr** [koreanbots/koreanbots#450] [제안] [answered]
## 제안 

v3 API에서는 디스코드와 같이 WebSocket API와 REST API를 같이 사용하는게 어떨까 합니다. 

## 필요성

업데이트 되었는지 확인하기 위해 일정 주기마다 다시 요청하는 것보다 WebSocket로 통신해 하트가 눌렸다면 바로 알릴 수 있어 API 서버가 부담하는 트래픽이 줄어드며,
봇 개발자들도 이를 통해서 개발자들은 SDK를 통해 이벤트로 받아와 다양한 기능을 만들 수 있습니다.

## 활용 방법

```js
client.koreanbots.on("vote", (user, bot) => {
    // ...
})
```

```py
@koreanbots.has_voted
@commands.command(name="ㅁㄴㅇㄹ")
async def test(self, ctx):
    return await ctx.send("하트 눌러주셔서 감사합니다!")
```

## 관련 참고 자료

* [WebSocket](https://npmjs.com/package/ws)

**Comment by wonderlandpark** [comment, marked as answer]
웹소켓은 v2에는 아직 구현할 생각이 없지만 Webhook으로 이벤트를 수신하는 경우엔 v2에 추가할 의사가 있습니다
